providers/base/units/watchdog/jobs.pxu

```diff
@@ -1,43 +1,137 @@
+_description:
+    Detect the watchdog is under the /sys/class/watchdog/ path and no other type of watchdog
 flags: simple
 imports: from com.canonical.plainbox import manifest
-requires: manifest.has_hardware_watchdog == 'True'
-command: udev_resource.py -f WATCHDOG
+requires:
+    manifest.has_hardware_watchdog == 'True'
+environ: WATCHDOG_TYPE WATCHDOG_IDENTITY
+command:
+    watchdog_config_test.py --detect
 
-id: watchdog/systemd-config
-_summary: Check if the hardware watchdog is properly configured
-template-engine: jinja2
-command: watchdog_config_test.py
+
+id: watchdog/set-timeout
 category_id: com.canonical.plainbox::power-management
+_summary: Configure the timeout for Hardware Watchdog
+_description:
+    Configure the value of RuntimeWatchdogSec
 flags: simple
 imports: from com.canonical.plainbox import manifest
-requires: manifest.has_hardware_watchdog == 'True'
-
-id: watchdog/trigger-system-reset-auto
-depends: watchdog/systemd-config
-_summary: Test that the watchdog module can trigger a system reset
+requires:
+    manifest.has_hardware_watchdog == 'True'
+    image_source_and_type.source == 'stock'
+depends:
+    watchdog/check-service
+    watchdog/detect
+user: root
 command:
-  sync
-  sleep 5
-  echo 1 > /proc/sys/kernel/sysrq
-  echo 0 > /proc/sys/kernel/panic
-  echo c > /proc/sysrq-trigger
-flags: preserve-locale noreturn autorestart
+    watchdog_config_test.py --set-timeout ${WATCHDOG_TIMEOUT:-35}
+
+
+id: watchdog/revert-timeout
+category_id: com.canonical.plainbox::power-management
+_summary: Restore the timeout for Hardware Watchdog
+_description:
+    Restore the value of RuntimeWatchdogSec
+flags: simple
+imports: from com.canonical.plainbox import manifest
+requires:
+    manifest.has_hardware_watchdog == 'True'
+    image_source_and_type.source == 'stock'
+depends:
+    watchdog/set-timeout
 user: root
+command:
+    watchdog_config_test.py --revert-timeout ${WATCHDOG_TIMEOUT:-35}
+
+
+id: watchdog/trigger-system-reset-auto
 plugin: shell
 category_id: com.canonical.plainbox::power-management
+_summary: Test that the watchdog module can trigger a system reset
+user: root
+flags: noreturn autorestart
 estimated_duration: 60
+depends:
+    watchdog/check-service
+    watchdog/detect
+command:
+    sync
+    sleep 5
+    echo 1 > /proc/sys/kernel/sysrq
+    echo 0 > /proc/sys/kernel/panic
+    echo c > /proc/sysrq-trigger
+
 
 id: watchdog/post-trigger-system-reset-auto
-after: watchdog/trigger-system-reset-auto
+plugin: shell
 category_id: com.canonical.plainbox::power-management
 _summary: Post watchdog reset service check
 _description: Check there are no failed services after the watchdog triggered
-unit: job
-plugin: shell
-command: failed_service_check.sh
 estimated_duration: 1.0
 imports: from com.canonical.plainbox import manifest
-requires: manifest.has_hardware_watchdog == 'True'
+requires:
+    manifest.has_hardware_watchdog == 'True'
+depends:
+    watchdog/trigger-system-reset-auto
```

Comment:
Because if `watchdog/trigger-system-reset-auto` fails, then it means the system did not reboot successfully because of system hang, then there is no need to check the failed service.